### PR TITLE
Add sign language speech translation

### PR DESCRIPTION
**Pull request by priya05-git**
📌 Description
This PR adds a complete, end-to-end ASL (American Sign Language) to Speech translation system to ML-CaPsule. The pipeline converts hand gestures captured via webcam into text and synthesized speech in real time, using MediaPipe for landmark extraction and a custom Transformer encoder built with PyTorch.
 What's Added
File	Description
preprocessing.py	Extracts 21 MediaPipe hand landmarks (63-D feature vectors), normalizes them, and builds a CSV dataset from image folders
train.py	Defines SignLanguageTransformer with CLS-token pooling, sinusoidal positional encoding, mixed-precision training (AMP), cosine LR scheduling, and early stopping
inference.py	Real-time webcam inference with confidence thresholding, prediction smoothing (rolling window), sentence accumulation, and TTS output via gTTS / pyttsx3
app.py	Gradio web app with 4 tabs: Image Upload, Live Webcam, Model Metrics, and About
requirements.txt	All required Python dependencies
README.md	Full documentation — architecture diagram, quick-start guide, dataset setup, training hyperparameters, inference options, and results

🏗️ Architecture
Webcam → MediaPipe (21 landmarks → 63-D vector)
       → SignLanguageTransformer (4× encoder layers, 8 attention heads, d_model=128)
       → Predicted Label + Confidence
       → Sentence Accumulator → gTTS / pyttsx3 → 🔊 Speech

✅ Key Features
🎯 ~97–99% validation accuracy on the ASL Alphabet Kaggle dataset (29 classes)
⚡ ~25–30 FPS on CPU, ~60 FPS on GPU
🪶 ~1.8 MB model size — suitable for CPU-only deployment
Supports both single-frame (alphabet) and temporal sequence (word-level) recognition
Attention heatmap visualization for interpretability
Data augmentation: landmark noise injection + rotation

🧪 How to Test
bash
pip install -r requirements.txt
# Preprocess dataset
python preprocessing.py --dataset_root dataset/asl_alphabet_train/ --output dataset/features.csv
# Train
python train.py --epochs 80 --d_model 128 --nhead 8 --num_layers 4
# Run web app
python app.py  # → http://localhost:7860
# Real-time webcam
python inference.py --camera 0 --threshold 0.80

📎 Related
Dataset: [ASL Alphabet — Kaggle](https://www.kaggle.com/datasets/grassknoted/asl-alphabet)
Contributes to: GSSoC '26

### CODE REVIEW

**Comment by Sibam-Paul** [comment]
first I can't find the issue no @priya05-git 
second there is already a project in the repo
https://github.com/Niketkumardheeryan/ML-CaPsule/tree/master/hand_sign_project
if u want to contribute ..u can make changes in the existing one or make another issue about a new concept or model
which is not there

**Comment by priya05-git** [comment]
Issue no. #1618.
Thanks for the clarification @Sibam-Paul ! I missed the existing project. I'll look for another concept that's not already in the repository and open a new issue before starting work.

**Comment by Sibam-Paul** [comment]
closing it not required
